sql-plugin/src/main/scala/org/apache/spark/sql/rapids/HashFunctions.scala

```diff
@@ -44,6 +44,26 @@ case class GpuMd5(child: Expression)
   }
 }
 
+case class GpuSha1(child: Expression)
+  extends GpuUnaryExpression with ImplicitCastInputTypes with NullIntolerantShim {
+  override def toString: String = s"sha1($child)"
+  override def inputTypes: Seq[AbstractDataType] = Seq(BinaryType)
+  override def dataType: DataType = StringType
+
+  override def doColumnar(input: GpuColumnVector): ColumnVector = {
+    withResource(HashUtils.normalizeInput(input.getBase)) { normalized =>
+      // at the moment sha1 on CuDF doesn't support lists, so have to translate
+      // BinaryType into StringType first before we operate on it
+      withResource(GpuCast.castToString(normalized, BinaryType, CastOptions.DEFAULT_CAST_OPTIONS))
```

Comment:
I am not sure that this is the API we want to use. Right now it does the correct thing, but there is no guarantee int he future it will. Especially if we start to try and detect bad UTF-8 character sequences and replace them in the same way that java does. It would probably be better to just copy the code here directly.

https://github.com/NVIDIA/spark-rapids/blob/2610f633b18dea377e8714e86b38a49a308560d7/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala#L1557-L1564

It is not that large.